specs/networking/p2p-interface.md

```diff
@@ -231,23 +230,24 @@ Request/response messages MUST adhere to the encoding specified in the protocol
 
 ```
 request   ::= <encoding-dependent-header> | <encoded-payload>
-response  ::= <result> | <encoding-dependent-header> | <encoded-payload>
+response  ::= <response_chunk>+
+response_chunk  ::= <result> | <encoding-dependent-header> | <encoded-payload>
 result    ::= “0” | “1” | “2” | [“128” ... ”255”]
 ```
 
 The encoding-dependent header may carry metadata or assertions such as the encoded payload length, for integrity and attack proofing purposes. Because req/resp streams are single-use and stream closures implicitly delimit the boundaries, it is not strictly necessary to length-prefix payloads; however, certain encodings like SSZ do, for added security.
 
-`encoded-payload` has a maximum byte size of `REQ_RESP_MAX_SIZE`.
+A `response` is formed by one or more `response_chunk`s. The exact request determines whether a response consists of a single `response_chunk` or possibly many. Responses that consist of a single SSZ-list (such as `BlocksByRange` and `BlocksByRoot`) send each list item as a `response_chunk`. All other response types (non-Lists) send a single `response_chunk`. The encoded-payload of a `response_chunk` has a maximum uncompressed byte size of `REQ_RESP_MAX_SIZE`.
```

Comment:
This can incur in excessive fragmentation/overhead/underutilisation if list items are small, as you're effectively defining a 1:1 mapping between chunk and list element.

I'd recommend to make the boundaries of chunks full list elements, allowing a chunk to contain multiple _complete_ list elements, and disallowing partial elements or bleeding over between chunks.

It may be worth formally introducing the term `chunkable`, so you can label message fields that satisfy this property in the schemas below.

Reply:
Please note that adding chunks does not increase the size of the overall payload, because we are replacing the leading 4-byte offsets that would appear in the SSZ lists before with response codes and varint lengths which will often amount to just 2 bytes per chunk.